### PR TITLE
feat(plugin-adr): use fetchApi instead of raw fetch

### PR DESCRIPTION
**Pull request by tcardonne**
## Hey, I just made a Pull Request!

In the ADR frontend plugin, use fetchApi instead of raw fetch in order to pass auth header if necessary.
This makes the ADR plugin compatible with the [Authenticate API Requests tutorial](https://github.com/backstage/backstage/blob/master/contrib/docs/tutorials/authenticate-api-requests.md)

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] ~Added or updated documentation~
- [ ] ~Tests for new functionality and regression tests for bug fixes~
- [ ] ~Screenshots attached (for UI changes)~
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by Depado — APPROVED** [review]

**Review by kuangp — APPROVED** [review]
Thanks!

File: .changeset/nice-schools-remember.md
Comment (kuangp) [inline .changeset/nice-schools-remember.md:2]: Just realized this should be `minor` since we're changing the `AdrClient` interface

**Review by kuangp — CHANGES_REQUESTED** [review]
Don't forget to run `yarn build:api-reports` locally to update api reports and commit changes

File: .changeset/nice-schools-remember.md
Comment (tcardonne) [reply .changeset/nice-schools-remember.md]: Alright, I've update the changeset of #17215 as well to match the change :+1:

File: .changeset/mean-rice-beg.md
Comment (kuangp) [inline .changeset/mean-rice-beg.md:2]: This package wasn't changed? did you include this accidentally when generating your changeset?

**Review by kuangp — CHANGES_REQUESTED** [review]

File: .changeset/mean-rice-beg.md
Comment (tcardonne) [reply .changeset/mean-rice-beg.md:2]: I intentionally changed this changeset which I created in #17215 for a similar PR which was merged.
Thought it made sense to change patch to minor here as well but I can remove this change if you want?

Comment (kuangp) [reply .changeset/mean-rice-beg.md:2]: hmm I'll let @jhaals decide how to proceed here, the issue would be that the changeset will link to the wrong commit sha in the resulting changelog

**Review by kuangp — APPROVED** [review]
Other than the extra changeset which I'll leave up to the maintainers to review, change looks great! thanks!

**Comment by tcardonne** [comment]
Thanks @kuangp for the review, I've removed the commit that modifies the Vault plugin changeset to avoid confusion in the changelog. :+1: 

I hope this PR will be included in the upcoming 1.13 release! :crossed_fingers:

**Review by jhaals — CHANGES_REQUESTED** [review]
Just a minor tweak to the changeset and then this is good to merge! :)

File: .changeset/nice-schools-remember.md
Comment (kuangp) [reply .changeset/nice-schools-remember.md:2]: Hey @jhaals 👋 

I actually suggested this be a `minor` since this is technically a breaking change due to the constructor of the public `AdrClient` interface being changed

Comment (jhaals) [reply .changeset/nice-schools-remember.md]: Hey @kuangp shoot, you're right!

File: .changeset/nice-schools-remember.md
Comment (jhaals) [inline .changeset/nice-schools-remember.md]:
```suggestion
'@backstage/plugin-adr': minor
```

**Review by jhaals — APPROVED** [review]

**Pull request merged**
